src/firebase/authService.js

```diff
@@ -0,0 +1,48 @@
+import firebase from '@firebase/app';
+import '@firebase/auth';
+import userService from './userService';
+
+const authService = function() {
+    let auth = null;
+
+    function init() {
+        auth = firebase.auth();
+        auth.getRedirectResult().then(result => {
+            if(result && result.user) {
+                userService.addUser(result.user);
+            }
+        })
+
+        firebase.auth().onAuthStateChanged(user => {
+            if(user) {
+                console.log(user);
+            } else {
+                console.log('no user');
+            }
+        })
+    }
+
+    function getCurrentUser() {
+        return auth.currentUser;
+    }  
+
+    function signIn() {
+        const provider = new firebase.auth.GoogleAuthProvider();
+        auth.signInWithRedirect(provider)
+    }
+
+    function signOut() {
+        auth.signOut();
+    }
+
+    return {
+        init,
+        getCurrentUser,
+        signIn,
+        signOut
+    }
+}
+
+const instance = authService();
```

Comment:
Any reason for this instead of 
`export default authservice;`

Reply:
This method makes the auth service a singleton. There is only one instance for the entire app.

In other components, you could have different instances of a component throughout the app.